examples/hosts/app-integration/live-schema-upgrade/src/modelVersion2/containerCode.ts

```
@@ -0,0 +1,139 @@
+		if (currentVersion === latestVersion) {
+			// We shouldn't try to upgrade if we are already on the latest version.
+			return;
+		}
+		console.log(`Upgrading to ${latestVersion}`);
+		if (this.container.connectionState !== ConnectionState.Connected) {
+			await new Promise((resolve) => {
+				this.container.once("connected", resolve);
+			});
+		}
+		const proposal: IFluidCodeDetails = { package: latestVersion };
+		this.container
+			.proposeCodeDetails(proposal)
+			.then((accepted: boolean) => {
+				console.log(`Upgrade accepted: ${accepted}`);
+			})
+			.catch((error) => {
+				console.error("Failed to upgrade:", error);
+			});
+	}
+}
+
+/**
+ * The runtime factory for our Fluid container.
+ */
+export class DiceRollerContainerRuntimeFactory extends ModelContainerRuntimeFactory<IDiceRollerAppModel> {
+	constructor() {
+		super(
+			new Map([
+				DiceRollerInstantiationFactory.registryEntry,
+				DiceCounterInstantiationFactory.registryEntry,
+			]), // registryEntries
+		);
+	}
+
+	/**
+	 * {@inheritDoc ModelContainerRuntimeFactory.containerInitializingFirstTime}
+	 */
+	protected async containerInitializingFirstTime(runtime: IContainerRuntime) {
+		const diceRoller = await runtime.createDataStore(DiceRollerInstantiationFactory.type);
+		await diceRoller.trySetAlias(diceRollerId);
+		const diceCounter = await runtime.createDataStore(DiceCounterInstantiationFactory.type);
+		await diceCounter.trySetAlias(diceCounterId);
+	}
+
+	/**
+	 * {@inheritDoc ModelContainerRuntimeFactory.createModel}
+	 */
+	protected async createModel(runtime: IContainerRuntime, container: IContainer) {
+		const diceRoller = await requestFluidObject<IDiceRoller>(
+			await runtime.getRootDataStore(diceRollerId),
+			"",
+		);
+		// Note: Since at this point is unclear whether or not this is the first time the app is being loaded with the
+		// new model, we should try to get the DiceCounter object and if it doesn't exist, create it.
+		let diceCounter: IDiceCounter;
+		try {
```

Ouch, the try/catch is annoying.  Probably the best option we have right now as I see getDataStore likes to throw rather than return undefined, but maybe something to think about improving.